staging/src/k8s.io/client-go/discovery/restmapper.go

```diff
@@ -108,6 +109,7 @@ func NewRESTMapper(groupResources []*APIGroupResources, versionInterfaces meta.V
 				plural := gv.WithResource(resource.Name)
 				singular := gv.WithResource(resource.SingularName)
 				versionMapper.AddSpecific(gv.WithKind(resource.Kind), plural, singular, scope)
+				versionMapper.AddSpecific(gv.WithKind(strings.ToLower(resource.Kind)), plural, singular, scope)
```

Comment:
@deads2k This avoids the aforementioned hack of "treating the kind as if it were a resource". versionMapper.AddSpecific maps a plural and singular resource name to a GVK object.

All that this is doing is mapping singular/plural resource -> GVK (to the gvk as-is)
as well as mapping singular / plural resource -> GVk (only kind in all lower-case)